egs/ami/s5/local/chain/run_tdnn_ami_4.sh

```diff
@@ -0,0 +1,308 @@
+where "nvcc" is installed.
+EOF
+fi
+
+# The iVector-extraction and feature-dumping parts are the same as the standard
+# nnet3 setup, and you can skip them by setting "--stage 8" if you have already
+# run those things.
+
+# if we are using the speed-perturbed data we need to generate
+# alignments for it.
+local/nnet3/run_ivector_common.sh --stage $stage \
+                                  --mic $mic \
+                                  --use-ihm-ali $use_ihm_ali \
+                                  --use-sat-alignments true || exit 1;
+
+
+# Set the variables. These are based on variables set by run_ivector_common.sh
+gmm=tri4a
+if [ $use_ihm_ali == "true" ]; then
+  gmm_dir=exp/ihm/$gmm
+  mic=${mic}_cleanali
+  ali_dir=${gmm_dir}_${mic}_train_parallel_sp_ali
+  lat_dir=${gmm_dir}_${mic}_train_parallel_sp_lats
+else
+  gmm_dir=exp/$mic/$gmm
+  ali_dir=${gmm_dir}_${mic}_train_sp_ali
+  lat_dir=${gmm_dir}_${mic}_train_sp_lats
+fi
+
+dir=exp/$mic/chain/${exp_name}${affix:+_$affix} # Note: _sp will get added to this if $speed_perturb == true.
+dir=${dir}_sp # we do speed perturbation by default, however for sake of consistency with older naming convention
+              # we  add _sp
+
+
+treedir=exp/$mic/chain/tri5_2y_tree_sp
+lang=data/$mic/lang_chain_2y
+
+final_lm=`cat data/local/lm/final_lm`
+LM=$final_lm.pr1-7
+graph_dir=$dir/graph_${LM}
+
+train_set=train_sp
+latgen_train_set=train_sp
+if [ $use_ihm_ali == "true" ]; then
+  latgen_train_set=train_parallel_sp
+fi
+
+###################################
+
+if [ $stage -le 10 ]; then
+  # Create a version of the lang/ directory that has one state per phone in the
+  # topo file. [note, it really has two states.. the first one is only repeated
+  # once, the second one has zero or more repeats.]
+  rm -rf $lang
+  cp -r data/lang $lang
+  silphonelist=$(cat $lang/phones/silence.csl) || exit 1;
+  nonsilphonelist=$(cat $lang/phones/nonsilence.csl) || exit 1;
+  # Use our special topology... note that later on may have to tune this
+  # topology.
+  steps/nnet3/chain/gen_topo.py $nonsilphonelist $silphonelist >$lang/topo
+fi
+
+if [ $stage -le 11 ]; then
+  # Build a tree using our new topology.
+  steps/nnet3/chain/build_tree.sh --frame-subsampling-factor 3 \
+      --leftmost-questions-truncate $leftmost_questions_truncate \
+      --cmd "$train_cmd" 4200 data/$mic/$latgen_train_set $lang $ali_dir $treedir
+
+  # this is done to prevent accidental overwrites of the tree
+  chmod -R 555 $treedir
+fi
+
+
+
+# combining the segments in training data to have a minimum length of frames_per_eg + tolerance
+# this is critical stage in AMI (gives 1% absolute improvement)
+if [ $stage -le 12 ]; then
+  min_seg_len=$(python -c "print ($frames_per_eg+5)/100.0")
+  steps/cleanup/combine_short_segments.py --minimum-duration $min_seg_len \
+    --input-data-dir data/$mic/${train_set}_hires \
+    --output-data-dirdata/$mic/${train_set}_min${min_seg_len}_hires
```

Comment:
Yes, there should be one. I running this setup now, will cleanup all bugs
after it is done.

On Fri, Mar 25, 2016 at 2:25 AM, Daniel Galvez <notifications@github.com>
wrote:

> In egs/ami/s5/local/chain/run_tdnn_ami_4.sh
> <https://github.com/kaldi-asr/kaldi/pull/619#discussion_r57422709>:
>
> > +      --leftmost-questions-truncate $leftmost_questions_truncate \
> > +      --cmd "$train_cmd" 4200 data/$mic/$latgen_train_set $lang $ali_dir $treedir
> > +
> > +  # this is done to prevent accidental overwrites of the tree
> > +  chmod -R 555 $treedir
> > +fi
> > +
> > +
> > +
> > +# combining the segments in training data to have a minimum length of frames_per_eg + tolerance
> > +# this is critical stage in AMI (gives 1% absolute improvement)
> > +if [ $stage -le 12 ]; then
> > +  min_seg_len=$(python -c "print ($frames_per_eg+5)/100.0")
> > +  steps/cleanup/combine_short_segments.py --minimum-duration $min_seg_len \
> > +    --input-data-dir data/$mic/${train_set}_hires \
> > +    --output-data-dirdata/$mic/${train_set}_min${min_seg_len}_hires
>
> Does this really work? I'm positive there should be a space right after
> "--output-data-dir"
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/619/files/44305e092c73e4c561b634e3160b7884464b5b43#r57422709>
>